src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -461,10 +461,23 @@ private void parseDoc(Map<String, String> namespacePrefixesByUri) {
                 final String instanceId = instanceNodeIdStrs.get(instanceIndex);
                 final String instanceSrc = instance.getAttributeValue(null, "src");
 
-                if (instanceSrc != null && instanceSrc.toLowerCase().startsWith("jr://file")) { // file or file-csv
+                final String externalSrc;
+                if (instanceSrc == null) {
+                    // It's internal, not external.
+                    externalSrc = null;
+                } else if (instanceSrc.toLowerCase().startsWith("jr://file")) { // file or file-csv
+                    externalSrc = instanceSrc;
+                } else if (instanceSrc.toLowerCase().startsWith("jr://instance/last-saved")) {
```

Comment:
Should this be `equals` instead of `startsWith`?

Reply:
I think you're right, @zestyping